### PR TITLE
 Replaced kfdefs with DataScienceCluster resource and changed default ns to opendatahub

### PR DESCRIPTION
**Pull request by hbelmiro**
## The issue resolved by this Pull Request:
Resolves https://github.com/opendatahub-io/data-science-pipelines-operator/issues/453

## Description of your changes:

Replaced kfdefs with DataScienceCluster resource and changed default ns to opendatahub

## Testing instructions

In a cluster with ODH Operator with 'fast' channel installed/ODH operator version > 2.0, create the following `DataScienceCluster`:

```bash
$ cat <<EOF | oc apply -f -
kind: DataScienceCluster
apiVersion: datasciencecluster.opendatahub.io/v1
metadata:
  name: data-science-pipelines-operator
  namespace: opendatahub
spec:
  components:
    dashboard:
      managementState: Managed
    datasciencepipelines:
      managementState: Managed
      devFlags:
        manifests:
          - uri: https://github.com/hbelmiro/data-science-pipelines-operator/tarball/issue-453
            contextDir: config
            sourcePath: base
EOF
```

The data-science-pipelines-operator should start successfully.

```bash
$ oc get pods -n opendatahub   
NAME                                                              READY   STATUS    RESTARTS   AGE
data-science-pipelines-operator-controller-manager-66b79b6lzrrz   1/1     Running   0          29s
odh-dashboard-f9c79d89b-k44pf                                     2/2     Running   0          33s
odh-dashboard-f9c79d89b-wz4xb                                     2/2     Running   0          33s
```

## Checklist
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by dsp-developers** [comment]
A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-483`
An OCP cluster where you are logged in as cluster admin is required.

To use this image run the following:

```bash
cd $(mktemp -d)
git clone git@github.com:opendatahub-io/data-science-pipelines-operator.git
cd data-science-pipelines-operator/
git fetch origin pull/483/head
git checkout -b pullrequest 0b06b02b8398b108c0dff1319efbfb09dab70bf3
make deploy IMG="quay.io/opendatahub/data-science-pipelines-operator:pr-483"
```

More instructions [here](https://github.com/opendatahub-io/data-science-pipelines-operator#deploy-dsp-instance) on how to deploy and test a Data Science Pipelines Application.

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-483`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-483`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-483`

**Comment by dsp-developers** [comment]
Change to PR detected. A new PR build was completed.
  A new image has been built to help with testing out this PR: `quay.io/opendatahub/data-science-pipelines-operator:pr-483`

File: .github/workflows/upgrade-test.yml [hunk 1]
```diff
@@ -88,20 +88,14 @@ jobs:
...
-      - name: Prepare for Upgrade Testing
-        run: |
-          # Update the KfDef manifest with the latest released version
-          sed -i "s/main/${{ inputs.released-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
-        working-directory: ${{ env.RESOURCES_DIR }}/kfdef
-
-      - name: Print KfDef Manifest Contents
-        run: cat ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
+      - name: Print DataScienceCluster Manifest Contents
+        run: cat ${{ env.RESOURCES_DIR }}/datasciencecluster/datasciencecluster.yaml
```
Comment (DharmitD) [inline .github/workflows/upgrade-test.yml:92]: The workflow replaces the branch in the kfdef uri to the released version here. You might need to add this step back as this is where the operator upgrade happens, but instead just replace `kfdef` with the `datasciencecluster` CR like you've done everywhere else.

File: .github/workflows/upgrade-test.yml [hunk 2]
```diff
@@ -120,20 +114,14 @@ jobs:
           DSPA_NS: test-ds-project-1
           WORKING_DIR: ${{ github.workspace }}
 
-      - name: Prepare for Upgrade Testing
-        run: |
-          # Update the KfDef manifest with the candidate version
-          sed -i "s/${{ inputs.released-version }}/${{ inputs.candidate-version }}/" ${{ env.RESOURCES_DIR }}/kfdef/kfdef.yaml
-        working-directory: ${{ env.RESOURCES_DIR }}/kfdef
+      - name: Print DataScienceCluster Manifest Contents
+        run: cat ${{ env.RESOURCES_DIR }}/datasciencecluster/datasciencecluster.yaml
```
Comment (DharmitD) [inline .github/workflows/upgrade-test.yml:118]: Same as https://github.com/opendatahub-io/data-science-pipelines-operator/pull/483/commits/745d8dd18e97da9d13e9ae446379c2f83dbef73e#r1405373939

Comment (hbelmiro) [reply .github/workflows/upgrade-test.yml:130]: Restored.

**Review by DharmitD — APPROVED** [review]
/Approve

**Pull request merged**